dex/elrond_dex_farm/src/lib.rs

```diff
@@ -343,29 +365,130 @@ pub trait Farm:
             .into())
     }
 
+    #[payable("*")]
+    #[endpoint(compoundRewards)]
+    fn compound_rewards(
+        &self,
+        #[payment_token] payment_token_id: TokenIdentifier,
+        #[payment_amount] payment_amount: Self::BigUint,
+        #[payment_nonce] payment_token_nonce: Nonce,
+        #[var_args] opt_accept_funds_func: OptionalArg<BoxedBytes>,
+    ) -> SCResult<CompoundRewardsResultType<Self::BigUint>> {
+        require!(self.is_active(), "Not active");
+        require!(payment_amount > 0, "Zero amount");
+
+        require!(!self.farm_token_id().is_empty(), "No issued farm token");
+        let farm_token_id = self.farm_token_id().get();
+        require!(payment_token_id == farm_token_id, "Unknown farm token");
+
+        let farming_token = self.farming_token_id().get();
+        let reward_token = self.reward_token_id().get();
+        require!(
+            farming_token == reward_token,
+            "Farming token differ from reward token"
+        );
+        self.generate_aggregated_rewards(&reward_token);
+
+        let current_rps = self.reward_per_share().get();
+        let farm_attributes = self.get_farm_attributes(&payment_token_id, payment_token_nonce)?;
+        let reward = self.calculate_reward(
+            &payment_amount,
+            &current_rps,
+            &farm_attributes.reward_per_share,
+        );
+
+        if reward > 0 {
+            self.decrease_reward_reserve(&reward)?;
+        }
+
+        let farm_token_id = self.farm_token_id().get();
```

Comment:
I think that farm_token_id already has this value (see line 381)

Reply:
Shouldn't be a problem

Reply:
Resolved